tests/unit/models/gpflow/test_models.py

```diff
@@ -432,46 +597,339 @@ def test_gaussian_process_regression_trajectory_sampler_has_correct_samples(
+
+
+def test_sparse_gaussian_process_regression_update_updates_num_data() -> None:
+    x_np = np.arange(5, dtype=np.float64).reshape(-1, 1)
+    x = tf.convert_to_tensor(x_np, x_np.dtype)
+    y = fnc_3x_plus_10(x)
+    m = SparseGaussianProcessRegression(sgpr_model(x, y))
+    num_data = m.model.num_data.numpy()
+
+    x_new = tf.concat([x, [[10.0], [11.0]]], 0)
+    y_new = fnc_3x_plus_10(x_new)
+    m.update(Dataset(x_new, y_new))
+    new_num_data = m.model.num_data.numpy()
+
+    assert new_num_data - num_data == 2
+
+
+def test_sparse_gaussian_process_regression_optimize_with_defaults() -> None:
+    x_observed = np.linspace(0, 100, 100).reshape((-1, 1))
+    y_observed = _3x_plus_gaussian_noise(x_observed)
+    data = x_observed, y_observed
+    dataset = Dataset(*data)
+
+    model = SparseGaussianProcessRegression(sgpr_model(x_observed, y_observed))
+    loss = model.model.training_loss()
+    model.optimize(dataset)
+
+    assert model.model.training_loss() < loss
+
+
+def test_sparse_gaussian_process_regression_optimize(compile: bool) -> None:
+    x_observed = np.linspace(0, 100, 100).reshape((-1, 1))
+    y_observed = _3x_plus_gaussian_noise(x_observed)
+    data = x_observed, y_observed
+    dataset = Dataset(*data)
+
+    optimizer = Optimizer(gpflow.optimizers.Scipy(), compile=compile)
+    model = SparseGaussianProcessRegression(sgpr_model(x_observed, y_observed), optimizer=optimizer)
+    loss = model.model.training_loss()
+    model.optimize(dataset)
+
+    assert model.model.training_loss() < loss
+
+
+@random_seed
+def test_sparse_gaussian_process_regression_trajectory_sampler_has_correct_samples() -> None:
+    x = tf.constant(np.arange(5).reshape(-1, 1), dtype=gpflow.default_float())
+    model = SparseGaussianProcessRegression(sgpr_model(x, _3x_plus_gaussian_noise(x)))
+    model.model.likelihood.variance.assign(1e-3)
+    model.update_posterior_cache()
+
+    num_samples = 300
+    trajectory_sampler = model.trajectory_sampler()
+
+    assert isinstance(trajectory_sampler, DecoupledTrajectorySampler)
+
+    trajectory = trajectory_sampler.get_trajectory()
+    x_predict = tf.constant([[1.0], [2.0], [3.0], [1.5], [2.5], [3.5]], gpflow.default_float())
+    x_predict_parallel = tf.expand_dims(x_predict, -2)  # [N, 1, D]
+    x_predict_parallel = tf.tile(x_predict_parallel, [1, num_samples, 1])  # [N, B, D]
+    samples = trajectory(x_predict_parallel)  # [N, B]
+    sample_mean = tf.reduce_mean(samples, axis=1, keepdims=True)
+    sample_variance = tf.math.reduce_variance(samples, axis=1, keepdims=True)
+
+    true_mean, true_variance = model.predict(x_predict)
+
+    # test predictions approx correct away from data
+    npt.assert_allclose(sample_mean[3:] + 1.0, true_mean[3:] + 1.0, rtol=0.2)
+    npt.assert_allclose(sample_variance[3:], true_variance[3:], rtol=0.5)
+
+    # test predictions almost correct at data
+    npt.assert_allclose(sample_mean[:3] + 1.0, true_mean[:3] + 1.0, rtol=0.1)
```

Comment:
for some reason had to relax the precision here and increase the number of samples above, not sure why - @henrymoss any thoughts?

Reply:
that looks  bit worrying! What was it before?

Reply:
for `test_sparse_variational_trajectory_sampler_has_correct_samples` we had  `num_samples = 100` and `rtol` here was `0.01`

Reply:
although I don't do this part in SGPR since it doesnt have q objects:
```
model.model.q_sqrt.assign(tf.expand_dims(tf.eye(len(x)) * tf.math.sqrt(1e-5), 0))
model.model.q_mu.assign(y)
model.update_posterior_cache()
```

Reply:
Is it the sample mean or sample variance test that has been relaxed?

Reply:
fixed it, all good